src/error.ts

```diff
@@ -33,6 +33,41 @@ export class FirebaseError extends Error {
   }
 }
 
+/**
+ * Safely gets an error message from an unknown object
+ * @param err an unknown error type
+ * @param defaultMsg an optional message to return if the err is not Error or string
+ * @return An error string
+ */
+export function getErrMsg(err: unknown, defaultMsg?: string): string {
```

Comment:
Love this -  those `unsafe any access` warnings on every `catch(err: any)` are very obnoxious